Add Salt for configuration management

maximiliankolb opened the pull request:
This PR adds a chapter on how to use Salt for configuration management. This is similar to [using Puppet for configuration management](https://docs.theforeman.org/nightly/Managing_Hosts/index-foreman-el.html#using-puppet-for-configuration-management_managing-hosts) (see 97b0a7d74ad36db27e99c6f1b5234ee4a39c1627).

melcorr commented on guides/common/modules/proc_salt-authenticating-salt-minions-using-salt-autosign-grains.adoc:
```diff
@@ -0,0 +1,56 @@
+----
+# systemctl restart salt-master
+----
+. Enable the Salt reactors and runners in your Salt Environment:
```
```suggestion
. Enable the Salt reactors and runners in your Salt environment:
```

maximiliankolb replied:
No change.
This is Salt terminology: `Salt Environment`, `Salt Variable`, and `Salt States`.

melcorr commented on guides/common/modules/proc_salt-configuring-salt-master.adoc:
```diff
@@ -0,0 +1,60 @@
+ext_pillar:
+  - puppet: /usr/bin/foreman-node
+----
+. Add a Salt Environment called `base` which is associated with the `/srv/salt` directory:
```
```suggestion
. Add a Salt Environment called `base` which is associated with the `/srv/salt` directory:
```

I corrected the E in another comment. Is this a term in Salt ?

maximiliankolb replied:
Yes, it's Salt terminology.

melcorr commented on guides/common/modules/proc_salt-configuring-salt-master.adoc:
```diff
@@ -0,0 +1,60 @@
+ext_pillar:
+  - puppet: /usr/bin/foreman-node
+----
+. Add a Salt Environment called `base` which is associated with the `/srv/salt` directory:
```
```suggestion
. Add a Salt Environment called `base` that is associated with the `/srv/salt` directory:
```

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-configuring-salt.adoc:
```diff
@@ -0,0 +1,16 @@
+[id="salt_guide_configuring_salt_{context}"]
+= Configuring Salt
+
+Once you've installed the Salt plugin, you need to connect it to a Salt Master.
```
I find this stuff interesting, I am unsure if anyone else cares, but for translation, this can be helpful for machine translation. This is from IBM Style: 

once adv
Use only to mean “one time.” Do not use as a conjunction to mean “after” or “when.”

In my previous job, I had a lovely book https://www.oreilly.com/library/view/the-global-english/9781599946573/ that explained things well with regards to technical translation. I am unsure if anyone cares and consider this a non-blocking comment also.

lzap replied:
I need to really read IBM Style now.

maximiliankolb replied:
Replaced with `after`.

melcorr commented on guides/common/modules/proc_salt-configuring-salt.adoc:
```diff
@@ -0,0 +1,16 @@
+ifdef::orcharhino[]
+[NOTE]
+====
+If you select Salt during step five of the xref:sources/installation_and_maintenance/orcharhino_installation_guide.adoc#oig_main_installation_steps[main orcharhino installation steps], the installer automatically performs those steps on your {Project}.
```
Are you sure you want to lock yourself into a step number?

maximiliankolb replied:
Removed `step five of`.

melcorr commented on guides/common/modules/proc_salt-creating-a-host-group-with-salt.adoc:
```diff
@@ -0,0 +1,19 @@
+. In the {ProjectWebUI}, navigate to *Configure > Host Groups*.
+. Click *Create Host Group*.
+.. On the *Host Group* tab, select a *Salt Environment* and a *Salt Master*.
+.. On the *Salt States* tab, assign *Salt States* to your host group.
```
```suggestion
.. Click the *Salt States* tab, and assign *Salt States* to your host group.
```
I'm assuming the Host Group tab is open by default? If not, maybe conform it to this style also.

maximiliankolb replied:
Host Group tab is open by default.

Applied to all three "on the X tab" sentences.

melcorr commented on guides/common/modules/proc_salt-creating-a-host-group-with-salt.adoc:
```diff
@@ -0,0 +1,19 @@
+ifdef::katello,orcharhino[]
+Managed hosts deployed using this host group automatically install and configure the required Salt Minion client software and register with your Salt Master.
+endif::[]
+See {ManagingHostsDocURL}creating-a-host-group[Creating a Host Group] in the _Managing Hosts_ guide for more information.
```
```suggestion
For more information, see {ManagingHostsDocURL}creating-a-host-group[Creating a Host Group] in the _Managing Hosts_ guide.
```

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-deploying-minion-hosts.adoc:
```diff
@@ -0,0 +1,17 @@
+Deploy managed hosts that are fully provisioned and configured for Salt usage.
+
+.Prerequisites
+. You have set up a _Salt Master_ and _Salt Environment_.
```
If we were to follow minimalist principles, this would be rewritten as something like. 

Before you can deploy Minion hosts, ensure that you have:
* A Salt Master...
* A Content VIew 
Etc

maximiliankolb replied:
Reworded.

melcorr commented on guides/common/modules/proc_salt-deploying-minion-hosts.adoc:
```diff
@@ -0,0 +1,17 @@
+.Prerequisites
+. You have set up a _Salt Master_ and _Salt Environment_.
+ifdef::katello,orcharhino[]
+. You have set up a content view containing the required Salt client software.
```
```suggestion
. You have set up a Content View containing the required Salt client software.
```

melcorr commented on guides/common/modules/proc_salt-importing-salt-states.adoc:
```diff
@@ -0,0 +1,22 @@
+
+A Salt State configures parts of a host, for example a service or the installation of a package.
+You can import Salt States from your Salt Master to {Project}.
+The Salt Master configuration used above uses a Salt Environment called `base` which includes the Salt States stored in `/srv/salt/`.
```
```suggestion
The Salt Master configuration in this guide uses a Salt Environment called `base` that includes the Salt States stored in `/srv/salt/`.
```

From IBM: 
above adj
Do not use to indicate a relative location in a document, as in “the above restrictions.” Use “previous” or
“preceding.”
Example: “The previous restrictions apply to...”

lzap replied:
Ok seriously I started looking up the IBM guide but its not in our README. Then I found its a book, like a real book on paper and Amazon (PDF). It is not free. Red Hat documentation team provides some extra information, but it looks to be complementary to the IBM guide: https://redhat-documentation.github.io/supplementary-style-guide/

Is my understanding correct that to efficiently contribute content, you need to buy the book?

melcorr replied:
It's a well known professional standard that most enterprise level software technical writers follow, especially in large software companies. Myself and @maximiliankolb have discussed it before and I think he was getting a copy for ATIX. 
I would only ever review on this level for @maximiliankolb because it is his profession and also because we've talked about it. Any Red Hat writers should have a copy of this and to use it. 
I wouldn't hold an engineer to these standards or expect them to look it up. I would expect a tech writer reviewing to suggest or implement the changes. 
This is where the professional writer/editor quality standards would come in. This was part of the original agreement between upstream and downstream - that these standards would be upheld. 
Maximilian has always shown interest and also tried to comply to the best of his ability, so I take the time to explain. 

TL;DR - I do not expect a contributor to know this, but a tech writer reviewing, should.

maximiliankolb replied:
Applied.

I totally get what @lzap is getting at: contributing to free software should ideally not depend on resources not accessible to everyone. I got my copy and refer to it whenever I can.

I am thankful for your/everyone's review and comments and I try my best to address them all.

melcorr commented on guides/common/modules/proc_salt-importing-salt-states.adoc:
```diff
@@ -0,0 +1,22 @@
+
+.Procedure
+. In the {ProjectWebUI}, navigate to *Configure > Salt States* and click *Import from FQDN*.
+** Click *Edit* to assign Salt States to Salt Environments.
```
These steps are optional? Why bullets?

maximiliankolb replied:
Removed the bullet points and yes, both steps are optional.

melcorr commented on guides/common/modules/proc_salt-importing-salt-states.adoc:
```diff
@@ -0,0 +1,22 @@
+This only removes the Salt State from {Project}, not from the disk of your Salt Master.
+. Click *Submit* to import the Salt States.
+
+Once imported, you can assign those Salt States to hosts or host groups.
```
See the previous comment on "once"

melcorr commented on guides/common/modules/proc_salt-installing-salt-plugin.adoc:
```diff
@@ -0,0 +1,28 @@
+[id="salt_guide_installing_salt_plugin_{context}"]
+= Installing Salt Plugin
+
+In order to configure managed hosts with Salt, you need to install the Salt plugin.
```
```suggestion
To configure managed hosts with Salt, you need to install the Salt plugin.
```
Minimalism.

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-running-salt.adoc:
```diff
@@ -0,0 +1,50 @@
+This includes an optional test mode that tells you what would happen without actually changing anything.
+
+.Procedure
+. Navigate to *Hosts > All Hosts*
```
In the {ProjectWebUI},etc

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-running-salt.adoc:
```diff
@@ -0,0 +1,50 @@
+. Click *Schedule Remote Job*.
+. For the *Job category* field, select _Salt_ from the drop down menu.
+. For the *Job template* field, select _Salt Run function_ from the drop down menu.
+. Fill in the function you want to trigger, for example `pillar.items` or `test.ping`.
```
```suggestion
. Complete the function you want to trigger, for example `pillar.items` or `test.ping`.
```
Normally, you'd mention the name of the field name. I see this applies to both, but is there any label on the text field?

maximiliankolb replied:
The field is called `function`. Should I keep the capitalization?

melcorr replied:
Obvs non-blocking but the way I would write it is:
"In the *Function* field, enter the name of the function that you want to trigger, for example, `pillar.items` or `test.ping`."

Just I used that Graphical User Interface Elements chapter of IBM style as a guide for so long. But this is fine.

melcorr replied:
Function should be bold not italics in my example ^

maximiliankolb replied:
Thanks, applied.

melcorr commented on guides/common/modules/proc_salt-running-salt.adoc:
```diff
@@ -0,0 +1,50 @@
+. Click *Submit* to run the job.
+
+You can also schedule remote jobs or run them recurrently.
+Refer to the {ManagingHostsDocURL}configuring-and-setting-up-remote-jobs_managing-hosts[Configuring and Setting up Remote Jobs] in the _managing hosts_ guide for more information.
```
```suggestion
For more information, see the {ManagingHostsDocURL}configuring-and-setting-up-remote-jobs_managing-hosts[Configuring and Setting up Remote Jobs] in the _managing hosts_ guide.
```

This is an IBM standard for references.

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-salt-example.adoc:
```diff
@@ -0,0 +1,51 @@
+. Add a new Salt State called `motd` to your Salt Master.
+.. Create the `/srv/salt/motd/` directory.
+.. Add two files to it:
+... The `/srv/salt/motd/init.sls` file is a Salt State file:
```
We never go this deep. 
You know, I see no value in indenting substeps if the user has to do them sequentially. 
I would appreciate if there isn't three levels of indentation. I'd rather no sub steps, since it looks to me that the user has to work through these all, so the hierarchy follows the UI rather the user, and we are writing user-oriented docs. 
Up to you though.

maximiliankolb replied:
Good call, fixed.

melcorr commented on guides/common/modules/proc_salt-salt-example.adoc:
```diff
@@ -0,0 +1,51 @@
+# salt '*' pillar.get vendor_name
+----
++
+If the output does not include the value of the `vendor_name` parameter, you need to refresh the Salt Pillar data first:
```
```suggestion
If the output does not include the value of the `vendor_name` parameter, you must refresh the Salt Pillar data first:
```

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-salt-example.adoc:
```diff
@@ -0,0 +1,51 @@
+# salt '*' saltutil.refresh_pillar
+----
++
+Refer to {ManagingHostsDocURL}salt_guide_enc_parameters[Salt ENC parameters] on how to refresh the pillar data.
```
```suggestion
For information about how to refresh pillar data, see to {ManagingHostsDocURL}salt_guide_enc_parameters[Salt ENC parameters].
```

Seems pillar isn't capitalised here and it is in other places, can you ctrl + f and make it one way or the other consistently?

maximiliankolb replied:
Applied and capitalized `Salt Pillar`.

melcorr commented on guides/common/modules/proc_salt-salt-variables.adoc:
```diff
@@ -0,0 +1,7 @@
+Similar to Puppet smart class parameters, you can configure Salt Variables within {Project}.
+The configured values are available as Salt Pillar data.
+
+Refer to the {ManagingHostsDocURL}puppet_guide_configuring_a_puppet_class_managing[Puppet smart class parameters] section on how to configure the variables' data.
```
For more information, see

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-using-content-management.adoc:
```diff
@@ -0,0 +1,23 @@
+[id="salt_guide_using_content_management_{context}"]
+= Using {Project}s Content Management
```
I would never know what to expect here.

maximiliankolb replied:
Should it say `Providing Salt Minion Client Software Using {Project}s Content Management`? Is it too long?

melcorr commented on guides/common/modules/proc_salt-using-content-management.adoc:
```diff
@@ -0,0 +1,23 @@
+[id="salt_guide_using_content_management_{context}"]
+= Using {Project}s Content Management
+
+This section provides best practice recommendations on how to provide managed hosts with the required Salt Minion client software using {Project}.
```
It's a procedure though, not a recommendation list?

maximiliankolb replied:
Reworded.

melcorr commented on guides/common/modules/proc_salt-using-content-management.adoc:
```diff
@@ -0,0 +1,23 @@
+
+.Procedure
+. Create and synchronize the required products and repositories.
+.. Create a product called `Salt`.
```
You could link to content management docs on how to create a product?

maximiliankolb replied:
Added multiple links.

melcorr commented on guides/common/modules/proc_salt-using-content-management.adoc:
```diff
@@ -0,0 +1,23 @@
++
+You can find the _upstream URL_ for the Salt packages on the https://repo.saltproject.io/[official Salt package repository].
+The URL depends on both the Salt version and the operating system, for example `\https://repo.saltproject.io/py3/ubuntu/20.04/amd64/3003/`.
+.. Synchronize the previously created products.
```
I see no value in the substeps and would prefer everything on the same level.

maximiliankolb replied:
Removed the substeps.

melcorr commented on guides/common/modules/proc_salt-using-content-management.adoc:
```diff
@@ -0,0 +1,23 @@
+.. Add each of your operating system specific Salt content views to your main composite content view for that operating system and version.
+.. Publish a new version of the composite content view from the previous step.
+.. Promote the content view from the previous step to your lifecycle environments as appropriate.
+.. Optional: Create activation keys for your composite content view and lifecycle environment combinations.
```
I think links would probably be helpful throughout this section. 
```suggestion
.. Optional: Create activation keys for your composite Content View and lifecycle environment combinations.
```

maximiliankolb replied:
Fixed capitalization and added links.

melcorr commented on guides/common/modules/proc_salt-using-content-management.adoc:
```diff
@@ -0,0 +1,23 @@
+.. Publish a content view for each of your content views from the previous step.
+. Create a composite content views for each major version of each operating system to make the new content available.
+.. Add each of your operating system specific Salt content views to your main composite content view for that operating system and version.
+.. Publish a new version of the composite content view from the previous step.
```
```suggestion
.. Publish a new version of the composite Content View from the previous step.
```

maximiliankolb replied:
Fixed.

melcorr commented on guides/common/modules/proc_salt-using-content-management.adoc:
```diff
@@ -0,0 +1,23 @@
+.. Create a content view for each repository.
+.. Publish a content view for each of your content views from the previous step.
+. Create a composite content views for each major version of each operating system to make the new content available.
+.. Add each of your operating system specific Salt content views to your main composite content view for that operating system and version.
```
```suggestion
.. Add each of your operating system specific Salt content views to your main composite Content View for that operating system and version.
```

melcorr commented on guides/common/modules/proc_salt-using-content-management.adoc:
```diff
@@ -0,0 +1,23 @@
+. Create and publish the required content views.
+.. Create a content view for each repository.
+.. Publish a content view for each of your content views from the previous step.
+. Create a composite content views for each major version of each operating system to make the new content available.
```
```suggestion
. Create a composite Content Views for each major version of each operating system to make the new content available.
```

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-using-content-management.adoc:
```diff
@@ -0,0 +1,23 @@
+.. Synchronize the previously created products.
+. Create and publish the required content views.
+.. Create a content view for each repository.
+.. Publish a content view for each of your content views from the previous step.
```
```suggestion
.. Publish a content view for each of your Content Views from the previous step.
```

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-using-content-management.adoc:
```diff
@@ -0,0 +1,23 @@
+You can find the _upstream URL_ for the Salt packages on the https://repo.saltproject.io/[official Salt package repository].
+The URL depends on both the Salt version and the operating system, for example `\https://repo.saltproject.io/py3/ubuntu/20.04/amd64/3003/`.
+.. Synchronize the previously created products.
+. Create and publish the required content views.
```
```suggestion
. Create and publish the required Content Views.
```

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-viewing-salt-autosign-keys.adoc:
```diff
@@ -0,0 +1,16 @@
+====
+
+.Procedure
+. In the {ProjectWebUI}, navigate to **.
```
??

melcorr replied:
What's happening here, friend? We can't really publish this without causing confusion, can we?

maximiliankolb replied:
Added the missing procedure.

melcorr reviewed (changes requested):
Hey @maximiliankolb 
Thanks! I wrote a few comments. 
Most are optional. I tried to explain the logic. 
Let me know if you need anything.

melcorr commented on guides/common/modules/con_salt-architecture.adoc:
```diff
@@ -0,0 +1,18 @@
+
+In Salt terminology, managed hosts are referred to as _Salt Minions_.
+
+{Team} recomends using the same Salt version on your Salt Minions as you are using on your Salt Master.
```
Recommend is mispelled here also.

maximiliankolb replied:
Removed `recommend` and reworded.

melcorr commented on guides/common/modules/con_salt-architecture.adoc:
```diff
@@ -0,0 +1,18 @@
+In Salt terminology, managed hosts are referred to as _Salt Minions_.
+
+{Team} recomends using the same Salt version on your Salt Minions as you are using on your Salt Master.
+Therefore, we also recommend using {Project}'s content management to provide managed hosts with the correct version of the Salt client software.
```
Typo here too.

maximiliankolb replied:
Reworded.

lzap commented on guides/common/modules/con_salt-architecture.adoc:
```diff
@@ -0,0 +1,18 @@
+For more information on installing a Salt Master, consult the https://docs.saltproject.io/en/latest/[official Salt documentation].
+Alternatively, use the bootstrap instructions found in the https://repo.saltproject.io/[official Salt package repository].
+
+[TIP]
+====
+{Team} recommends using the same Salt version on the Salt Master as you are using on your Salt Minions.
+====
```
Mel was very patient with many my `[NOTE]` statements.

@melcorr would you mind explaining this in the README why this is not wanted? I think it is pretty easy to avoid this, we just need to know.

maximiliankolb replied:
Removed `tip` and reworded `recommend`.

FYI: recommended is used a lot.

lzap commented on guides/common/modules/proc_salt-authenticating-salt-minions-using-salt-autosign-grains.adoc:
```diff
@@ -0,0 +1,56 @@
+  runner.foreman_https.query_cert:
+    - method: PUT
+    - host: {foreman-example-com}
+    - path: /salt/api/v2/salt_autosign_auth?name={{ data['id'] }}
```
This might get replaced with asciidoc attributes, I havent seen the result tho. Might be ok.

maximiliankolb replied:
This does not seem to be a problem:

<img width="997" alt="attributes" src="https://user-images.githubusercontent.com/12595287/133099506-bd3b3083-9b1b-4ee2-808c-a790f05ee71e.png">

lzap commented on guides/common/modules/proc_salt-configuring-foreman.adoc:
```diff
@@ -0,0 +1,35 @@
+====
+The command `adduser saltuser -p password` does not work.
+Using it prevents you from importing Salt States.
+====
```
Warning: WARNING :-)

maximiliankolb replied:
I've wrapped it in `ifdef`. Is that okay for you?

lzap commented on guides/common/modules/proc_salt-importing-salt-states.adoc:
```diff
@@ -0,0 +1,22 @@
+====
+Configure the paths for Salt States and Salt Pillars in `/etc/salt/master`.
+By default, Salt States are located in `/srv/salt` and and Salt Pillars in `/srv/pillar`.
+====
```
Warning: NOTE

maximiliankolb replied:
It's a note. Does this need to be changed?

melcorr replied:
No :)

lzap commented on guides/common/modules/proc_salt-installing-salt-plugin.adoc:
```diff
@@ -0,0 +1,28 @@
+Select *Salt* as a configuration management system during step five of the xref:sources/installation_and_maintenance/orcharhino_installation_guide.adoc#oig_main_installation_steps[main orcharhino installation steps].
+Choosing this option installs and configures both the Salt plugin and a Salt Master on your orcharhino.
+====
+endif::[]
```
Warning: TIP

melcorr replied:
IMO: if he wraps it for orcharhino, he's free to do what he likes without comment from me :dancers:

maximiliankolb replied:
No change.

maximiliankolb commented on guides/common/assembly_using-salt-for-configuration-management.adoc:
```diff
@@ -0,0 +1,63 @@
+
+Use this section as a guide to configuring {Project} to use Salt for configuration management on managed hosts.
+
+include::modules/con_salt-instroduction.adoc[leveloffset=+1]
```
Fixed.

maximiliankolb commented on guides/common/modules/con_salt-architecture.adoc:
```diff
@@ -0,0 +1,18 @@
+[id="salt_guide_salt_architecture_{context}"]
+= Salt Architecture
+
+You need a Salt Master which either runs on your {ProjectServer} or {SmartProxyServer} with the Salt plugin enabled.
```
Applied.

maximiliankolb commented on guides/common/modules/con_salt-instroduction.adoc:
```diff
@@ -0,0 +1,17 @@
+Salt can be used as a configuration management tool similar to {ConfiguringAnsibleDocURL}using-ansible-roles_ansible[Ansible] or {ManagingHostsDocURL}using-puppet-for-configuration-management_managing-hosts[Puppet].
+
+This guide describes how to use Salt for configuration management in {Project}.
+It shows how to install the Salt plugin, how to integrate {Project} with an existing Salt Master, and how to configure managed hosts with Salt.
```
Applied.

maximiliankolb commented on guides/common/modules/proc_salt-activating-salt.adoc:
```diff
@@ -0,0 +1,31 @@
+[id="salt_guide_activating_salt_{context}"]
+= Activating Salt
+
```
Once Salt Master is installed, services need to be started and on Foreman, services need to be restarted. Afterwards, SmartProxy on Foreman needs to be refreshed to connect to Salt Master.

maximiliankolb commented on guides/common/modules/proc_salt-authenticating-salt-minions-using-host-names.adoc:
```diff
@@ -0,0 +1,33 @@
+= Authenticating Salt Minions Using Host Names
+
+Configure {Project} to authenticate Salt Minions based on their host names.
+This relies on the `autosign.conf` file which stores the host names of Salt Minions the Salt Master accepts.
```
Thanks! Applied.

maximiliankolb commented on guides/common/modules/proc_salt-authenticating-salt-minions-using-host-names.adoc:
```diff
@@ -0,0 +1,33 @@
+
+[NOTE]
+====
+{Team} recommends authenticating Salt Minions using Salt Grains.
```
Removed.

maximiliankolb commented:
Thanks for the review so far @melcorr & @lzap. I've addressed most issues by now and rebased to master.

melcorr commented:
I will review first thing tomorrow morning. 
Sorry about the delay with this and thank you so much for applying the feedback.

melcorr commented on guides/common/modules/proc_salt-configuring-salt-master.adoc:
```diff
@@ -0,0 +1,60 @@
+  ssl_key: /etc/puppetlabs/puppet/ssl/private_keys/{foreman-example-com}.pem
+  ssl_crt: /etc/puppetlabs/puppet/ssl/certs/{foreman-example-com}.pem
+----
+. Optional: Use Salt as remote execution provider.
```
```suggestion
. Optional: Use Salt as a remote execution provider.
```

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-configuring-salt-master.adoc:
```diff
@@ -0,0 +1,60 @@
+----
+. Optional: Use Salt as remote execution provider.
++
+You can run arbitrary commands on managed hosts relying on the connection to the Salt Minion.
```
I found this sentence hard to understand on rereading. Relying on = dependent on? It doesn't seem to make the most sense here. Connected to the Salt Minion? Maybe I just don't understand. 
```suggestion
You can run arbitrary commands on managed hosts relying on the connection to the Salt Minion.
```

maximiliankolb replied:
Reworded.
You can do REX without relying on SSH because Salt Master already has a connection to its Salt Minions.

melcorr commented on guides/common/modules/proc_salt-enabling-salt-grains-upload.adoc:
```diff
@@ -0,0 +1,27 @@
+= Enabling Salt Grains Upload
+
+Managed hosts running the Salt Minion client software can upload Salt Grains to {ProjectServer} or {SmartProxyServer}.
+Salt Grains are analogous to Puppet facts.
```
Can we always be so sure that a Salt user understands Puppet terminology? I would suggest that spelling out what we mean here would be safer.

maximiliankolb replied:
Fixed.

melcorr commented on guides/common/modules/proc_salt-creating-a-host-group-with-salt.adoc:
```diff
@@ -0,0 +1,19 @@
+.Procedure
+. In the {ProjectWebUI}, navigate to *Configure > Host Groups*.
+. Click *Create Host Group*.
+.. Click the *Host Group* tab and select a *Salt Environment* and a *Salt Master*.
```
```suggestion
. Click the *Host Group* tab and select a *Salt Environment* and a *Salt Master*.
```

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-creating-a-host-group-with-salt.adoc:
```diff
@@ -0,0 +1,19 @@
+. In the {ProjectWebUI}, navigate to *Configure > Host Groups*.
+. Click *Create Host Group*.
+.. Click the *Host Group* tab and select a *Salt Environment* and a *Salt Master*.
+.. Click the *Salt States* tab and assign *Salt States* to your host group.
```
```suggestion
. Click the *Salt States* tab and assign *Salt States* to your host group.
```

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-creating-a-host-group-with-salt.adoc:
```diff
@@ -0,0 +1,19 @@
+.. Click the *Host Group* tab and select a *Salt Environment* and a *Salt Master*.
+.. Click the *Salt States* tab and assign *Salt States* to your host group.
+ifdef::katello,orcharhino[]
+.. Click the *Activation Keys* tab and select an activation key containing the Salt Minion client software.
```
```suggestion
. Click the *Activation Keys* tab and select an activation key containing the Salt Minion client software.
```

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-importing-salt-states.adoc:
```diff
@@ -0,0 +1,22 @@
+[id="salt_guide_importing_salt_states_{context}"]
+= Importing Salt States
+
+A Salt State configures parts of a host, for example a service or the installation of a package.
```
```suggestion
A Salt State configures parts of a host, for example, a service or the installation of a package.
```

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-viewing-salt-autosign-keys.adoc:
```diff
@@ -0,0 +1,16 @@
+= Viewing Salt Autosign Keys
+
+The Salt Keys page lists hosts and their Salt keys.
+You can manually accept, reject, or delete keys:
```
```suggestion
You can manually accept, reject, or delete keys.
```

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-salt-variables.adoc:
```diff
@@ -0,0 +1,7 @@
+[id="salt_guide_salt_variables_{context}"]
+= Salt Variables
+
+Similar to Puppet smart class parameters, you can configure Salt Variables within {Project}.
```
```suggestion
You can configure Salt Variables within {Project}.
```
I think it's best not to create the illusion of dependency on Puppet knowledge

maximiliankolb replied:
Applied.

melcorr commented on guides/common/modules/proc_salt-salt-variables.adoc:
```diff
@@ -0,0 +1,7 @@
+Similar to Puppet smart class parameters, you can configure Salt Variables within {Project}.
+The configured values are available as Salt Pillar data.
+
+For more information, see {ManagingHostsDocURL}puppet_guide_configuring_a_puppet_class_managing[Puppet smart class parameters] section on how to configure the variables' data.
```
This link does not work for me?

maximiliankolb replied:
Fixed.

melcorr commented on guides/common/modules/proc_salt-viewing-enc-parameters.adoc:
```diff
@@ -0,0 +1,28 @@
+Click *Salt ENC* on the host overview page to view assigned Salt States.
+This shows a list of parameters that are made available for Salt usage as Salt Pillar data.
+
+You can check what parameters are truly available on the Salt side by running the following command on your Salt Master:
```
```suggestion
You can check what parameters are truly available on the Salt side by completing the following procedure.
```

maximiliankolb replied:
Applied.

melcorr reviewed (changes requested):
Hey @maximiliankolb 
One hard requirement: there is a section that looks incomplete. 
The rest are suggestions and non-blocking. 
These are very good docs. It has been a pleasure to review. 
Thanks to you and ATIX for the contribution. 
I would love to host a community deep dive sometime if you or your colleagues would be interested in showcasing this.

maximiliankolb commented:
Thanks again Melanie for your kind words and review! I've rebased to master, checked and fixed all cross references, and resolved all issues. I think there are two more threads open, so we're almost there.

melcorr reviewed (approved):
lgtm!

maximiliankolb commented:
> lgtm!

Thanks @melcorr 🥳
